src/util.c

```diff
@@ -419,28 +419,32 @@ static UNUSED CONST inline unsigned ctz64(uint64_t x)
 // We define these separately from `ctz32` etc. so that we can verify all of
 // `ctz32` etc. without necessarily linking them into the kernel binary.
 #ifdef CONFIG_CLZ_32
-CONST int __clzsi2(uint32_t x)
+compile_assert(clzsi2_int_is_32_bits, 4 == sizeof(unsigned int))
+CONST int __clzsi2(unsigned int x)
 {
     return clz32(x);
 }
 #endif
 
 #ifdef CONFIG_CLZ_64
-CONST int __clzdi2(uint64_t x)
+compile_assert(clzdi2_long_is_64_bits, 8 == sizeof(unsigned long))
```

Comment:
In fact, I think this is just plain wrong for all of our 32-bit platforms, and the only reason this compiles is that we don't yet have any platforms that define `CONFIG_CLZ_64`.

See the `clz_word_size` `compile_assert` further up, which is always on, and shows that `unsigned long` is 32 bits on our current 32-bit platforms.

Reply:
You are right, this is broken. Fixed the check.

Reply:
If my conviction that `si` is always 32 bits and `di` is always 64 bits turns out to be correct, then I think your update is in the wrong direction.